crypto/src/merkle_tree/proof.rs

```diff
@@ -0,0 +1,132 @@
+use crate::hash::traits::IsCryptoHash;
+use lambdaworks_math::{
+    errors::ByteConversionError,
+    field::{element::FieldElement, traits::IsField},
+    traits::ByteConversion,
+};
+
+pub struct Proof<F: IsField, H: IsCryptoHash<F>> {
+    pub value: FieldElement<F>,
+    pub merkle_path: Vec<(FieldElement<F>, bool)>,
+    pub hasher: H,
+}
+
+impl<F, H> ByteConversion for Proof<F, H>
+where
+    F: IsField,
+    H: IsCryptoHash<F>,
+    FieldElement<F>: ByteConversion,
+{
+    /// Returns the byte representation of the element in big-endian order.
+    fn to_bytes_be(&self) -> Vec<u8> {
+        let mut buffer: Vec<u8> = Vec::new();
+
+        for val in self.value.to_bytes_be().iter() {
+            buffer.push(*val);
+        }
+
+        for (value, is_left) in self.merkle_path.iter() {
+            for val in value.to_bytes_be().iter() {
+                buffer.push(*val);
+            }
+
+            if *is_left {
+                buffer.push(1);
+            } else {
+                buffer.push(0);
+            }
+        }
+
+        buffer.to_vec()
+    }
+
+    /// Returns the byte representation of the element in little-endian order.
+    fn to_bytes_le(&self) -> Vec<u8> {
+        let mut buffer: Vec<u8> = Vec::new();
+
+        for val in self.value.to_bytes_le().iter() {
+            buffer.push(*val);
+        }
+
+        for (value, is_left) in self.merkle_path.iter() {
+            for val in value.to_bytes_le().iter() {
+                buffer.push(*val);
+            }
+
+            if *is_left {
+                buffer.push(1);
+            } else {
+                buffer.push(0);
+            }
+        }
+
+        buffer.to_vec()
+    }
+
+    /// Returns the element from its byte representation in big-endian order.
+    fn from_bytes_be(bytes: &[u8]) -> Result<Self, ByteConversionError> {
+        let mut merkle_path = Vec::new();
+
+        for elem in bytes[8..].chunks(9) {
+            let field = FieldElement::from_bytes_be(&elem[..elem.len() - 1])?;
+            merkle_path.push((field, elem[elem.len() - 1] == 1));
+        }
+
+        Ok(Proof {
+            value: FieldElement::from_bytes_be(&bytes[..8])?,
+            merkle_path,
+            hasher: H::new(),
+        })
+    }
+
+    /// Returns the element from its byte representation in little-endian order.
+    fn from_bytes_le(bytes: &[u8]) -> Result<Self, ByteConversionError> {
+        let mut merkle_path = Vec::new();
+
+        for elem in bytes[8..].chunks(9) {
+            let field = FieldElement::from_bytes_le(&elem[..elem.len() - 1])?;
+            merkle_path.push((field, elem[elem.len() - 1] == 1));
+        }
+
+        Ok(Proof {
+            value: FieldElement::from_bytes_le(&bytes[..8])?,
+            merkle_path,
+            hasher: H::new(),
+        })
+    }
+}
+
+#[cfg(test)]
+
+mod tests {
```

Comment:
We could add a TODO here to add tests for the other methods of `ByteConversion`. After the TODO is added, we need to file an issue to track it.

Reply:
I could add those now it is trivial I think

Reply:
Ok, go ahead!